### PR TITLE
normalize the object space normal that gets passed to shadergraph

### PR DESCRIPTION
**Pull request by elizabeth-legros**
Bugfix for https://fogbugz.unity3d.com/f/cases/1330848/
original fix was part of https://github.com/Unity-Technologies/Graphics/pull/3467 but that pull was not going to be backported. 
Was not able to reproduce error after bugfix.

### CODE REVIEW

**Review by cdxntchou — APPROVED** [review]
Looks good

**Review by jessebarker — APPROVED** [review]
LGTM

File: com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/SharedCode.template.hlsl [hunk 1]
```diff
@@ -29,7 +29,7 @@ SurfaceDescriptionInputs FragInputsToSurfaceDescriptionInputs(FragInputs input,
     #if defined(SHADER_STAGE_RAY_TRACING)
     $SurfaceDescriptionInputs.ObjectSpaceNormal:         output.ObjectSpaceNormal =           mul(output.WorldSpaceNormal, (float3x3) ObjectToWorld3x4());
     #else
-    $SurfaceDescriptionInputs.ObjectSpaceNormal:         output.ObjectSpaceNormal =           mul(output.WorldSpaceNormal, (float3x3) UNITY_MATRIX_M);           // transposed multiplication by inverse matrix to handle normal scale
+    $SurfaceDescriptionInputs.ObjectSpaceNormal:         output.ObjectSpaceNormal =           normalize(mul(output.WorldSpaceNormal, (float3x3) UNITY_MATRIX_M));           // transposed multiplication by inverse matrix to handle normal scale
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/SharedCode.template.hlsl:32]: The raytracing one also need to be normalize, it is the same issue

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/SharedCode.template.hlsl:32]: I will do it

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/SharedCode.template.hlsl:32]: Thanks for doing this PR!

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**